src/realm/object-store/impl/collection_notifier.cpp

```diff
@@ -103,14 +103,14 @@ void CollectionNotifier::recalculate_key_path_array()
     m_any_callbacks_filtered = false;
     m_key_path_array.clear();
     for (const auto& callback : m_callbacks) {
-        if (callback.key_path_array.empty()) {
+        if (!callback.key_path_array) {
             m_all_callbacks_filtered = false;
         }
         else {
             m_any_callbacks_filtered = true;
-        }
-        for (const auto& key_path : callback.key_path_array) {
-            m_key_path_array.push_back(key_path);
+            for (const auto& key_path : callback.key_path_array.value()) {
```

Comment:
Use `*` rather than `.value()`. `value()` requires a higher deployment target on Apple platforms, and has somewhat weird semantics.